### ISSUE TITLE
Intermittent failure parsing sshd_config

### ISSUE THREAD

**Issue by snhughes** [PowerShell/Win32-OpenSSH#1496]
**Troubleshooting steps**
1. Add the following lines to sshd_config. Change the user W_ttt_scan if you need. 
```
Match Group administrators
    AuthorizedKeysFile __PROGRAMDATA__/ssh/administrators_authorized_keys
Match User W_ttt_scan
    PasswordAuthentication no
```
2. Restart sshd then try to connect to the server with the user (ex.: ssh W_ttt_scan@localhost). Sometimes it asks for the password and sometimes it does not. It should never ask for password.

**Expected output**
Never prompt for password

**Actual output**
Sometimes prompt for password

**Terminal issue? please go through wiki**
Don't think so.

**"OpenSSH for Windows" version**
 7.9.0.0
 
**Server OperatingSystem**
Windows Server 2012 R2 Standard

**Client OperatingSystem**
Windows Server 2012 R2 Standard

**What is failing**
Something is going wrong when parsing the sshd_config file.

If I move the 'Match User' line above the 'Match Group' line, the issue does not occur.

We can see to following lines in the log/debug output when is erroneously asks for a password  (note the 'n' user):
```
user n matched group list administrators at line 89
checking match for 'User W_tvm_scan' user n host ::1 addr ::1 laddr ::1 lport 22
```

When it correctly rejects the connection (due too certificate missing), this is the log/debug output (all as expected):
```
user w_tvm_scan matched group list administrators at line 89
checking match for 'User W_tvm_scan' user w_tvm_scan host ::1 addr ::1 laddr ::1 lport 22
```

**Some debugging results:**
There is no 'n' user in the system.
It says user 'n' even when you change the user to something without the letter 'n'.
If you change the 'AuthorizedKeysFile' line to another 'PasswordAuthentication' line, the issue goes away.
Adding new lines between the Match lines do not help.

**Comment by NoMoreFood** [comment]
There's a group matching bug in that version.  Please re-baseline with the latest.

Reference: https://github.com/PowerShell/openssh-portable/pull/380

**Comment by snhughes** [comment]
NoMoreFood, thanks for the reply.
However, at least with 8.0.0-p1 Beta release, the 'Match' doesn't seems to be working at all. I get a 'Match not found' for the user in the logs.
I don't know if it should be fixed on 8.0.0-p1 Beta or only on the latest master source code.
Let me know if you would like me to raise a new bug report for it.

**Comment by maertendMSFT** [comment]
Please provide the server-side and client-side logs
